src/spo2.c

```diff
@@ -0,0 +1,88 @@
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "ls_task.h"
+#include "spo2.h"
+#include "spo2_driver.h"
+
+
+ls_task spo2_task;
+
+
+/**
+ * do_spo2() - spo2 task worker function
+ * @pvParameters:	void* to pass function parameters.
+ *
+ * This worker function retrieves an input data sample from the ADCs. This
+ * is the place where further processing of the data will happen.
+ */
+static void do_spo2(void *pvParameters);
+
+/**
+ * spo2_read() - read a single input data sample
+ * @data:	space for sample readings
+ *
+ * Read out a single data sample using the different peripherals needed.
+ */
+static void spo2_read(uint32_t *data);
+
+/**
+ * print_values() - print aquired values
+ * @data:	space for sample readings
+ *
+ * Print values to serial out that have been aquired through the peripherals
+ * for demonstration and/or debug purpuses.
+ */
+static void print_values(uint32_t *data);
+
+
+void
+spo2
+(void)
+{
+	spo2_init_peripherals();
```

Comment:
why hide the init of each component in this wrapper?

Reply:
why should this pollute main?